packages/actionmenu/src/react/item.tsx

```diff
@@ -43,99 +47,107 @@ const styles = {
   textOnly: () => css(stylesheet['.psds-actionmenu__text-only'])
 }
 
-interface Props extends Omit<HTMLAttributes<HTMLLIElement>, 'onClick'> {
-  disabled?: boolean
-  tagName?: string
+interface ItemProps extends Omit<HTMLAttributes<HTMLLIElement>, 'onClick'> {
   className?: string
-  origin?: string
-  onClick?: (event: React.MouseEvent, value: string | number) => void
+  disabled?: boolean
+  href?: string
   nested?: React.ReactNode
+  onClick?: (event: React.MouseEvent, value: string | number) => void
+  origin?: string
+  tagName?: string
   value?: string | number
 }
-
-export const Item = forwardRef<HTMLLIElement, Props>(
-  (
-    {
+export const Item = forwardRef<HTMLLIElement, ItemProps>(
+  (props, forwardedRef) => {
+    const {
+      children,
+      className,
       disabled,
       nested,
+      onClick,
       origin,
+      tagName: TagName = tagName.a,
       value,
-      className,
-      children,
-      onClick,
-      tagName,
       ...rest
-    },
-    forwardedRef
-  ) => {
-    const { onClickContext, onClose, originContext } = useContext<{
-      onClickContext: (e: Event, v: string | number) => void
-      onClose: (e: Event, v: string | number) => void
-      originContext: string
-    }>(ActionMenuContext)
+    } = props
```

Comment:
i moved the destructuring to inside the fn...only because my editor code highlighting can't figure it out otherwise....personal problem...

Reply:
orly? That's surprising. This is very common in our code, including elsewhere in this file. What's special about this case? Also feels odd to make coding decisions based on code highlighting tools, but I can't be sure I haven't done the same thing. :)

Reply:
yeah...probably not the best approach from me

Reply:
If it helps, I think we destructure way too much in our code. I think in some cases it harms readability, requiring code tracing more often than not.  

Reducing destructuring to required cases might be another way to help.